src/uproot/writing/identify.py

```diff
@@ -761,15 +763,14 @@ def to_TList(data, name=""):
         )
 
     tobject = uproot.models.TObject.Model_TObject.empty()
-    tobject._members["@fUniqueID"] = 0
-    tobject._members["@fBits"] = 0
 
     tlist = uproot.models.TList.Model_TList.empty()
     tlist._bases.append(tobject)
     tlist._members["fName"] = name
     tlist._data = list(data)
     tlist._members["fSize"] = len(tlist._data)
-    tlist._options = [b""] * len(tlist._data)
+    # see https://github.com/scikit-hep/uproot5/pull/763#issuecomment-1290149135
+    tlist._options = [uproot.serialization.string("")] * len(tlist._data)
```

Comment:
I don't know what `_options` are used for, but they've always been `b""` in all the files/contexts I've seen. Maybe it is a used-to-be-used feature that is today ignored.

Reply:
> That's right: only the functions and modules with "serialize" or "write" in their names should be concerned with serialization/writing. identify.py is for generating `Interpretation` objects, to know how to map C++ objects in TTrees into data in arrays.
> 
> The `_options` should continue to be Python objects, `b""`, and only get serialized with the `bytestring` function when the serialization process is already underway (in a `serialize` method).

I implemented the requested change in d6e154308dee74c8a21170578e32af0f20827551 but left the `_options` as python `str`. In your comment you suggest keeping it as python `bytes` instead. It feels wierd to me that in the `serialize` method this is converted back into a `str` only to be serialized in a different way. Specially when other members such as `_members["fName"]` are `str`.

Any reason why the `_options` entries should be `bytes` and not `str` @jpivarski ?

Reply:
Yeah, I had no idea what they represented, whether they would ever be human-readable or not. That draw `option` in `TObject::SetDrawOption` is not necessarily related to this `option`. I've never seen what TList options do.

Strictly speaking, any `char*` or `std::string` from C++ ought to be a Python `bytes`: those are at the same level of abstraction. In Uproots 1‒3, they were, but users complained about the `bytes` handling, especially as Python 3 became the only Python.

So we have a policy of presenting human-readable/interpretable strings (like `fName` and `fTitle`) as `str` and binary stuff (like `fUUID`) as `bytes`. We use `"utf-8"` with `errors="surrogateescape"` as an encoding convention. It's not necessarily what the user intended—maybe they expect all of their axis labels to be in `"latin-1"`, or maybe `"latin-4"`, but UTF-8 is ASCII for ASCII and `errors="surrogateescape"` allows non-UTF-8 to at least be round-trippable:

```python
>>> b"he\xffllo".decode("utf-8")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 2: invalid start byte
```

but

```python
>>> b"he\xffllo".decode("utf-8", errors="surrogateescape")
'he\udcffllo'
```

and

```python
>>> b"he\xffllo".decode("utf-8", errors="surrogateescape").encode(errors="surrogateescape")
b'he\xffllo'
```

I think there are functions in _util.py for these conversions.

About your real question: wanting to avoid double-conversion. I just noticed that there's no `bytestring` function to complement our

https://github.com/scikit-hep/uproot5/blob/33777b16c654f1fa4c26a960af1e70f2357425d3/src/uproot/serialization.py#L19-L32

You can just add one (also in serialization.py):

```python
def bytestring(data):
    """
    Converts Python bytes into another byte sequence that is ready to be serialized
    into a file.

    If the length of the bytes object is less than 255, it is preceded by a 1-byte
    length; otherwise, it is preceded by ``b'\xff'`` and a 4-byte length.
    """
    length = len(data)
    if length < 255:
        return struct.pack(">B%ds" % length, length, data)
    else:
        return struct.pack(">BI%ds" % length, 255, length, data)
```

Reply:
Indeed it does! I opened up a browser window, went to the appropriate file in GitHub, and didn't see it there—in `main`. I should have thought of also checking _this branch_!

This comment thread can be resolved now because it's no longer mixing serialization with representation:

https://github.com/lobis/uproot5/blob/2d6917490b86a7cb89884d5b08e9bcb7ac0424e1/src/uproot/writing/identify.py#L772